pkg/virtctl/vm/stop.go

```diff
@@ -66,27 +66,29 @@ func (o *Command) stopRun(args []string, cmd *cobra.Command) error {
 		fmt.Printf("Dry Run execution\n")
 	}
 
-	if cmd.Flags().Changed(gracePeriodArg) && forceRestart == false {
-		return fmt.Errorf("Can not set gracePeriod without --force=true")
-	}
-
-	if forceRestart {
-		if cmd.Flags().Changed(gracePeriodArg) {
-			err = virtClient.VirtualMachine(namespace).ForceStop(context.Background(), vmiName, &v1.StopOptions{GracePeriod: &gracePeriod, DryRun: dryRunOption})
-			if err != nil {
-				return fmt.Errorf("Error force stopping VirtualMachine, %v", err)
-			}
-		} else if !cmd.Flags().Changed(gracePeriodArg) {
-			return fmt.Errorf("Can not force stop without gracePeriod")
+	if cmd.Flags().Changed(gracePeriodArg) {
```

Comment:
I think the complexity is still there, just in a bit different order.

To reduce complexity, you could treat the errors first and only then continue.
E.g.:
```go
gpChanged = cmd.Flags().Changed(gracePeriodArg)

if gpChanged != forceRestart {
  return fmt.Errorf("Can not set gracePeriod without `--force=true`" or force stop without gracePeriod")
}

if forceRestart {
  err = virtClient.VirtualMachine(namespace).ForceStop(context.Background(), vmiName, &v1.StopOptions{GracePeriod: &gracePeriod, DryRun: dryRunOption})
 ...
} else {
  err = virtClient.VirtualMachine(namespace).Stop(context.Background(), vmiName, &v1.StopOptions{DryRun: dryRunOption})
  ...
}

fmt.Printf("VM %s was scheduled to %s\n", vmiName, o.command)
return nil
```

Reply:
Done